compiler/noirc_driver/src/lib.rs

```diff
@@ -352,9 +352,10 @@ pub fn check_crate(
     let crate_files = context.crate_files(&crate_id);
     let warnings_and_errors: Vec<FileDiagnostic> = diagnostics
         .into_iter()
-        .map(|(error, file_id)| {
+        .map(|error| {
+            let location = error.location();
             let diagnostic = CustomDiagnostic::from(&error);
-            diagnostic.in_file(file_id)
+            diagnostic.in_file(location.file)
```

Comment:
Seems like we can get rid of `FileDiagnostic` now, no?
We get the file from the location here than attach the file to the outer diagnostic as well - seems unnecessary.

Reply:
The main issue is that `CustomDiagnostic` doesn't have a main file. All the files are in `secondaries` and that can be empty. There's `CustomDiagnostic::from_message` that ends up with a file, though in almost all places we immediately turn that into a `FileDiagnostic` with a file.

We can add a `file: FileId` field to `CustomDiagnostic` and make sure one is there whenever it's created. What do you think? I'm fine doing it in this PR or in a follow-up one.

Reply:
Ah, ok. It's not as straightforward as I thought then so lets leave it for a separate PR since this one is already rather large.